### PR TITLE
ws_health_exporter role was added

### PR DESCRIPTION
**Pull request by kogeler**
Changes:
* the ws_health_exporter role was published
* GitHub actions were improved to support multiple Ansible roles
* LXD feature was added to the GitHub actions. It allows us to be independent of host OS versions.

# LXD
Using Docker as the molecule driver you can't:
* use Ubuntu 22.04 as the GitHub action runner OS 
* run molecule for the node role on Debian 11

It's because if you need systemd inside a Docker container you will not be able to use different OS versions inside the container and the host (cgroup, systemd conflicts). Because Docker and Podman don't provide full isolation in these parts. LXD provides full OS isolation without virtualization.

### CODE REVIEW

File: roles/ws_health_exporter/molecule/default/group_vars/all.yml [hunk 1]
```diff
@@ -0,0 +1,22 @@
+## Molecule
+ansible_user: root
+
+## node
+node_binary_version: "v0.9.40"
```
Comment (aardbol) [inline roles/ws_health_exporter/molecule/default/group_vars/all.yml]: You're not basing it on the version, like you did in other roles?

File: roles/ws_health_exporter/molecule/default/molecule.yml [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+  name: galaxy
+driver:
+  name: ${DRIVER:-docker}
+platforms:
+  - name: molecule-instance-ws-health-exporter
+    # LXD
+    source:
+      alias: debian/bullseye/amd64
+    # DOCKER
+    image: "paritytech/debian10:latest"
```
Comment (aardbol) [inline roles/ws_health_exporter/molecule/default/molecule.yml]: Shouldn't this be debian 11 now?

**Review by aardbol — CHANGES_REQUESTED** [review]

File: roles/ws_health_exporter/templates/.service.j2 [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+Environment="WSHE_WS_CHECK_INTERVAL={{ ws_health_exporter_ws_check_interval }}"
+Environment="WSHE_WS_TIMEOUT={{ ws_health_exporter_ws_timeout }}"
+Environment="WSHE_NODE_MAX_UNSYNCHRONIZED_BLOCK_DRIFT={{ ws_health_exporter_node_max_unsynchronized_block_drift }}"
+Environment="WSHE_NODE_MIN_PEERS={{ ws_health_exporter_node_min_peers }}"
+
+ExecStart={{ _ws_health_exporter_venv }}/bin/python3 {{ _ws_health_exporter_file }}
+
+Restart=always
+User={{ ws_health_exporter_user }}
+Group={{ ws_health_exporter_user}}
```
Comment (aardbol) [inline roles/ws_health_exporter/templates/.service.j2]: Will this var work? Missing space in` {{ ws_health_exporter_user}}`

File: roles/ws_health_exporter/molecule/default/molecule.yml (hunk 2 above)
Comment (kogeler) [reply roles/ws_health_exporter/molecule/default/molecule.yml]: fixed

Comment (aardbol) [reply roles/ws_health_exporter/molecule/default/molecule.yml]: looking goooood

**Review by aardbol — APPROVED** [review]
lgtm

File: roles/ws_health_exporter/tasks/main.yml [hunk 4]
```diff
@@ -0,0 +1,79 @@
+---
+
+- block:
+
+  - name: ws_health_exporter | remove OLD exporter file
+    ansible.builtin.file:
+      path: "{{ _ws_health_exporter_old_file }}"
+      state: absent
```
Comment (BulatSaif) [inline roles/ws_health_exporter/tasks/main.yml:8]: It is better to remove this step, this is "new" role and git history is empty,  this step may confuse new users.

File: roles/ws_health_exporter/tasks/main.yml [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+      name:
+        - "prometheus-client==0.16.0"
+        - "websocket-client==1.5.1"
+        - "apscheduler==3.10.1"
+        - "flask==2.2.3"
+        - "environs==9.5.0"
+        - "waitress==2.1.2"
+      virtualenv: "{{ _ws_health_exporter_venv }}"
+      virtualenv_command: "python3 -m venv"
+    tags: ['ws-health-exporter-test1']
```
Comment (BulatSaif) [inline roles/ws_health_exporter/tasks/main.yml]:
```suggestion
```

File: roles/ws_health_exporter/vars/main.yml [hunk 6]
```diff
@@ -0,0 +1,7 @@
+---
+
+_ws_health_exporter_name: "ws-health-exporter"
+_ws_health_exporter_file: "{{ ws_health_exporter_base_path }}/exporter.py"
+_ws_health_exporter_venv: "{{ ws_health_exporter_base_path }}/venv"
+
+_ws_health_exporter_old_file: "/home/{{ ws_health_exporter_user }}/bin/{{ _ws_health_exporter_name }}.py"
```
Comment (BulatSaif) [inline roles/ws_health_exporter/vars/main.yml]:
```suggestion
```

**Review by BulatSaif — APPROVED** [review]
Good job, small cleanup needed

File: roles/ws_health_exporter/tasks/main.yml (hunk 4 above)
Comment (BulatSaif) [inline roles/ws_health_exporter/tasks/main.yml]:
```suggestion
```

It is better to remove this step, this is "new" role and git history is empty,  this step may confuse new users.

**Review by BulatSaif — APPROVED** [review]

**Pull request merged**